akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1790,6 +1791,27 @@ trait FlowOpsMat[+Out, +Mat] extends FlowOps[Out, Mat] {
     viaMat(alsoToGraph(that))(matF)
 
   /**
+   * Materializes to `Future[Unit]` that completes on getting termination message.
+   * The Future completes with success when received complete message from upstream or cancel
+   * from downstream. It fails with the same error when received error message from
+   * downstream.
+   *
+   * `watchTermination` ignores the current [[Flow]] materialization value, in other words it's combine
+   * values with Keep.right
+   */
+  def watchTermination(): ReprMat[Out, Future[Unit]] =
+    viaMat(GraphStages.terminationWatcher)(Keep.right)
```

Comment:
it will behave like proxy than. Is there a case where we need to stay with it? I mean should we keep Mat only method?